src/include/platform/OTARequestorInterface.h

```diff
@@ -180,9 +195,11 @@ class OTARequestorInterface
     // Get image update progress in percents unit
     virtual CHIP_ERROR GetUpdateProgress(EndpointId endpointId, chip::app::DataModel::Nullable<uint8_t> & progress) = 0;
 
-    // Get requestor state
+    // Get the value of the UpdateState attribute of the OTA Software Update Requestor Cluster on the given endpoint
     virtual CHIP_ERROR GetState(EndpointId endpointId,
                                 chip::app::Clusters::OtaSoftwareUpdateRequestor::OTAUpdateStateEnum & state) = 0;
+    // Getter for the value of the UpdateState cached by the object
+    virtual app::Clusters::OtaSoftwareUpdateRequestor::OTAUpdateStateEnum GetCurrentUpdateState() = 0;
```

Comment:
The APIs here tie platform layer to generated cluster logic, which is circular. It only works because the circularity is not yet made full, but this may happen in some builds or in the future. Please do not mix data model code generated types with "platform code". Extract the elements at play, which are all fundamentally in <lib/core/DataModelTypes.h> (lib/core/ is a dependency of Platform layer, whereas app has a depnedency ON platform layer). 

There are very few enums at play, and keeping these enums synced can be enforced by static asserts as done elsewhere.

Reply:
Done